app/models/bookmark.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+class Bookmark < ApplicationRecord
+  belongs_to :user, inverse_of: :bookmarks, optional: false
+  belongs_to :reply, inverse_of: :bookmarks, optional: false
+  belongs_to :post, inverse_of: :bookmarks, optional: false
+
+  validates :type, uniqueness: { scope: [:user, :reply] }
+  validates :type, inclusion: { in: ['reply_bookmark'] }, allow_nil: false
```

Comment:
Can `type` be an enum? https://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html

(Can still use string columns if desirable!)

Reply:
Uhhh I guess it could but other enum columns like editor_mode in reply are not and use the inclusion validation

Reply:
(Discussed aloud: keeping it as-is to maintain coherence with other tables, though we might want to look into it for all columns that are enumerable in the future.)